### PR TITLE
[ENG-2841] Delete Revision In Progress Button

### PR DESCRIPTION
**Pull request by chth0n1x**
-   Ticket: []
-   Feature flag: n/a

## Purpose

The purpose of this change was to enable users to delete revisions in progress.

## Summary of Changes

-Added a delete button to the registries-left-nav
-Added logic to the component.ts file to the registries-left-nav file to enable the delete

## Side Effects

This change may potentially conflict with other permission or moderations states and should be tested for each of the RegistrationReviewStates and RevisionReviewStates.

### CODE REVIEW

File: lib/osf-components/addon/components/osf-layout/registries-side-nav/component.ts [hunk 1]
```diff
@@ -31,4 +42,24 @@ export default class RegistriesSideNav extends Component {
     toggle() {
         this.toggleProperty('shouldCollapse');
     }
+
+    @action
+    delete() {
+        this.set('deleteEdit', false);
+        const node = this.toDelete;
```
Comment (adlius) [inline lib/osf-components/addon/components/osf-layout/registries-side-nav/component.ts:49]: Where do we set `this.toDelete` in the first place?

File: lib/osf-components/addon/components/osf-layout/registries-side-nav/component.ts [hunk 2]
```diff
@@ -4,16 +4,27 @@ import { action } from '@ember/object';
...
 @layout(template, styles)
 export default class RegistriesSideNav extends Component {
     @service media!: Media;
+    @service toast!: Toast;
+    @service intl!: Intl;
+
+    toDelete: SchemaResponseModel | null;
+    deleteEdit = false;
+
+    reloadList?: (page?: number) => void; // bound by paginated-list
```
Comment (adlius) [inline lib/osf-components/addon/components/osf-layout/registries-side-nav/component.ts:27]: What list are we reloading?

File: app/models/registration.ts [hunk 3]
```diff
@@ -104,10 +104,12 @@ export default class RegistrationModel extends NodeModel.extend(Validations) {
     @attr('fixstring') reviewsState!: RegistrationReviewStates;
     @attr('fixstring') iaUrl?: string;
     @attr('array') providerSpecificMetadata!: ProviderMetadata[];
-    @attr('fixstring') revisionState!: RevisionReviewStates;
+    @attr('fixstring') revisionState?: RevisionReviewStates;
+    @attr('object') latestSchemaResponse?: SchemaResponseModel;
 
     // Write-only attributes
     @attr('array') includedNodeIds?: string[];
+    @attr('boolean') createDoi?: boolean;
```
[the remark this answers is not in the document]
Comment (chth0n1x) [reply app/models/registration.ts:112]: This was from the most recent merge commit 5e3f4e9993966792b0844febddf1d9dd0742b357 to resolve conflicts. I believe I was asked to keep it, but should it be removed?

File: app/models/registration.ts [hunk 4]
```diff
@@ -104,10 +104,12 @@ export default class RegistrationModel extends NodeModel.extend(Validations) {
     @attr('fixstring') reviewsState!: RegistrationReviewStates;
     @attr('fixstring') iaUrl?: string;
     @attr('array') providerSpecificMetadata!: ProviderMetadata[];
-    @attr('fixstring') revisionState!: RevisionReviewStates;
+    @attr('fixstring') revisionState?: RevisionReviewStates;
+    @attr('object') latestSchemaResponse?: SchemaResponseModel;
```
[the remark this answers is not in the document]
Comment (chth0n1x) [reply app/models/registration.ts]: Ok that make sense, thank you Yuhuai. Would removing and committing those changes affect any other part of the code? I just looked and do not see it on registration.ts so should be good now.

File: lib/osf-components/addon/components/osf-layout/registries-side-nav/component.ts (hunk 1 above)
Comment (chth0n1x) [reply lib/osf-components/addon/components/osf-layout/registries-side-nav/component.ts]: Hey, after our discussion this morning this a lot more sense. I will wait to discuss with you later before removing this, to make sure the changes are how you envision.

File: lib/osf-components/addon/components/osf-layout/registries-side-nav/component.ts (hunk 2 above)
Comment (chth0n1x) [reply lib/osf-components/addon/components/osf-layout/registries-side-nav/component.ts:40]: This logic is not complete but was in place for if a list of revision ids were needed, the logic would reload the list and pull the last object of the schema id from the list after the deletion and redirect to its overview page.

**Comment by adlius** [comment]
Closed in favor of #1348

**Pull request closed without merging**